src/serde/de.rs

```diff
@@ -77,21 +31,8 @@ pub fn node_from_stream(allocator: &mut Allocator, f: &mut Cursor<&[u8]>) -> io:
                     ops.push(ParseOp::Cons);
                     ops.push(ParseOp::SExp);
                     ops.push(ParseOp::SExp);
-                } else if b[0] == 0x01 {
-                    values.push(allocator.one());
-                } else if b[0] == 0x80 {
-                    values.push(allocator.null());
-                } else if b[0] <= MAX_SINGLE_BYTE {
-                    values.push(allocator.new_atom(&b)?);
                 } else {
-                    let blob_size = decode_size(f, b[0])?;
-                    let pos = f.position() as usize;
-                    if f.get_ref().len() < pos + blob_size as usize {
-                        return Err(bad_encoding());
-                    }
-                    let blob = &f.get_ref()[pos..(pos + blob_size as usize)];
-                    f.seek(SeekFrom::Current(blob_size as i64))?;
-                    values.push(allocator.new_atom(blob)?);
+                    values.push(parse_atom(allocator, b[0], f)?);
```

Comment:
most of this patch is moving functions to new files, but this is one place where the code is refactored. Is it the only place, or is there some other places that warrant extra review too?

Reply:
I believe this is the only new non-test function.